apps/client/src/app/core/electron/packet-capture-tracker.service.ts

```diff
@@ -341,6 +342,29 @@ export class PacketCaptureTrackerService {
     this.ipc.playerSetupPackets$.subscribe((packet) => {
       this.userInventoryService.setContentId(BigInt(packet.contentID).toString(16).padStart(16, '0').toUpperCase());
     });
+
+    this.ipc.freecompanyId$.subscribe((packet) => {
+      this.freecompanyWorkshopFacade.setCurrentFreecompanyId(packet);
+    });
+
+
+    combineLatest([
+      this.ipc.freecompanyId$,
+      this.freecompanyWorkshopFacade.vesselPartUpdate$
+    ]).pipe(
```

Comment:
Having a `combineLatest` here means that if the FC id changes, the last vessel part update will me emitted along with it, which I don't think is the expected behavior 😄.

I think changing this for a switchMap approach based on the freeCompanyId$ observable would solve it, like (pseudocode ofc):

```
freeCompanyId$
  filter not null
  switchMapTo(vesselPartUpdates$)
  subscribe do your thing
```

Reply:
This is actually the expected behaviour for "testing purpose". Still thinking how I'll do it because the packets are only sent when you go to the workshop. I might just suscribe to InitZone packet and filter on the ZoneId and the latest freeCompanyId.

Reply:
Another way to do that would be to store the FC id inside the store, this way you can access it anytime. I don't know how often it's sent tho, so it's a bit hard to figure out how often it's updated by the server.

Reply:
I already have the FC id inside the store for most of the stuff. 
I might add a check before adding it to the store tho. Only store it if it's a FC that the user "imported". 

The FC packet is sent every time you load a zone or you join a FC. What I actually meant was that the ItemInfo packets that I need are only sent when you load the workshop zone.
So I might just do something like that

```
vesselPartUpdate$ (ItemInfo | UpdateInventorySlot)
  withLatestFrom currentFreeCompanyId$
     filter not null
```

No idea why I mentionned InitZone, I don't actually need it at all, maybe I was thinking of something else :thinking: